### PR TITLE
NOS29 - Follow functionality

### PR DESCRIPTION
**Pull request by cjorgensen**
- Adds follow button to profile header
- Adds a placeholder list of follows view
- Adds EVENT and CLOSE relay writes
- Removes some of the didSet requests to reduce number of REQs sent
- Renames Profile class to CurrentUser

### CODE REVIEW

File: Nos/Views/FollowsView.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        .onReceive(syncTimer.currentTimePublisher) { _ in
+            if !authorsToSync.isEmpty {
+                print("Syncing \(authorsToSync.count) authors")
+                let authorKeys = authorsToSync.map({ $0.hexadecimalPublicKey! })
+                let filter = Filter(authorKeys: authorKeys, kinds: [.metaData], limit: authorsToSync.count)
+                relayService.requestEventsFromAll(filter: filter)
+                authorsToSync.removeAll()
+            }
+        }
```
Comment (jcswift23) [inline Nos/Views/FollowsView.swift:53]: You can add:
.navigationBarBackButtonHidden(true) to hide the extra back button.
If you want the navigationTitle to say: "Follows" you can change the navigation title added code would be:
    .navigationBarBackButtonHidden(true)
    .onAppear {
      router.navigationTitle = "Follows"
    }

Comment (cjorgensen) [reply Nos/Views/FollowsView.swift:58]: Pushed this fix.  Thanks!

**Review by jcswift23 — APPROVED** [review]

File: Nos/Service/Filter.swift [hunk 2]
```diff
@@ -7,23 +7,22 @@
...
-    private var authorPublicKeys: [String]
-    var kinds: [EventKind]
-    var limit: Int
+    private var authorKeys: [String]
+    private var kinds: [EventKind]
+    private var limit: Int
 
-    init(publicKeys: [String] = [], kinds: [EventKind] = [], limit: Int = 100) {
-        self.authorPublicKeys = publicKeys
+    init(authorKeys: [String] = [], kinds: [EventKind] = [], pTags: [String] = [], limit: Int = 100) {
```
Comment (mplorentz) [inline Nos/Service/Filter.swift:16]: Looks like `pTags` is unused?

**Review by mplorentz — COMMENTED** [review]
This is awesome! I left some comments you can address in a future PR if you want.

**Pull request merged**